static/app/utils/profiling/hooks/useProfilingFunctionMetrics.tsx

```diff
@@ -0,0 +1,46 @@
+import {useMemo} from 'react';
+
+import {normalizeDateTimeParams} from 'sentry/components/organizations/pageFilters/parse';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import useOrganization from 'sentry/utils/useOrganization';
+import usePageFilters from 'sentry/utils/usePageFilters';
+
+interface UseProfilingFunctionMetricsProps {
+  fingerprint: Profiling.FunctionMetric['fingerprint'];
+  projects: number[];
+  query?: string;
+}
+
+export function useProfilingFunctionMetrics(props: UseProfilingFunctionMetricsProps) {
+  const organization = useOrganization();
+  const {selection} = usePageFilters();
+
+  const path = `/organizations/${organization.slug}/events-stats/`;
+  const endpointOptions = useMemo(() => {
+    const params: {
+      query: Record<string, any>;
+    } = {
+      query: {
+        project: props.projects,
+        environment: selection.environments ?? selection.environments,
+        ...normalizeDateTimeParams(selection.datetime ?? selection.datetime),
+        dataSource: 'profilesMetrics',
+        fingerprint: props.fingerprint,
+        query: props.query,
```

Comment:
I dont think it takes a fingerprint as a query param. Instead, you should be passing the fingerprint inside the query like
```suggestion
        query: `${props.fingerprint}`,
```